kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/ApplicationStream.java

```diff
@@ -0,0 +1,249 @@
+package com.wepay.kafka.connect.bigquery.write.storageApi;
+
+import com.google.cloud.bigquery.storage.v1.WriteStream;
+import com.google.cloud.bigquery.storage.v1.BigQueryWriteClient;
+import com.google.cloud.bigquery.storage.v1.JsonStreamWriter;
+import com.google.cloud.bigquery.storage.v1.FinalizeWriteStreamResponse;
+import com.google.cloud.bigquery.storage.v1.BatchCommitWriteStreamsRequest;
+import com.google.cloud.bigquery.storage.v1.BatchCommitWriteStreamsResponse;
+import com.google.cloud.bigquery.storage.v1.StorageError;
+import com.google.protobuf.Descriptors;
+import com.wepay.kafka.connect.bigquery.exception.BigQueryStorageWriteApiConnectException;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Plain JAVA class with all utility methods on Application streams.
+ * Streams which are created by calling application are called as Application streams
+ * Uses stream writer methods from:
+ * https://cloud.google.com/java/docs/reference/google-cloud-bigquerystorage/latest/com.google.cloud.bigquery.storage.v1
+ */
+public class ApplicationStream {
+
+    private static final Logger logger = LoggerFactory.getLogger(ApplicationStream.class);
+    private StreamState currentState = null;
+    private final String tableName;
+    private WriteStream stream = null;
+    private JsonStreamWriter jsonWriter = null;
+    private final Map<TopicPartition, OffsetAndMetadata> offsetInformation;
+    private final BigQueryWriteClient client;
+    /**
+     * Number of times append is called
+     */
+    private final AtomicInteger appendCalls;
+    /**
+     * Number of append requests completed successfully. This can never be greater than appendCalls
+     */
+    private final AtomicInteger completedCalls;
+
+    /**
+     * This is called by builder to guarantee sequence.
+     */
+    private final AtomicInteger maxCalls;
```

Comment:
Could you explain how this AtomicInteger is guaranteeing sequence? Also maxCalls does not convey the message that this is used to guarantee sequence - can we think of some other name? Thanks!

Reply:
MaxCalls just tells maximum calls this stream has to make. It does not manage sequence. The code path which takes us here and executes `updateOffsetInformation`  actually handles no missing offsets.

Reply:
Ah I see, the description might be mis-leading. I will fix it. Thanks for pointing it out.